libminifi/include/core/state/nodes/AgentInformation.h

```diff
@@ -425,77 +427,123 @@ class AgentStatus : public StateMonitorNode {
+    }
+    serialized.push_back(serializeUptime());
 
-    SerializedResponseNode uptime;
-
-    uptime.name = "uptime";
-    if (nullptr != monitor_) {
-      uptime.value = monitor_->getUptime();
-    } else {
-      uptime.value = "0";
+    auto serializedComponents = serializeComponents();
+    if (!serializedComponents.empty()) {
+      serialized.push_back(serializedComponents);
     }
 
-    if (!repositories_.empty()) {
-      SerializedResponseNode repositories;
+    serialized.push_back(serializeResourceConsumption());
 
-      repositories.name = "repositories";
+    return serialized;
+  }
+
+ protected:
+  SerializedResponseNode serializeRepositories() {
+    SerializedResponseNode repositories;
 
-      for (auto &repo : repositories_) {
-        SerializedResponseNode repoNode;
-        repoNode.collapsible = false;
-        repoNode.name = repo.first;
+    repositories.name = "repositories";
 
-        SerializedResponseNode queuesize;
-        queuesize.name = "size";
-        queuesize.value = repo.second->getRepoSize();
+    for (auto &repo : repositories_) {
+      SerializedResponseNode repoNode;
+      repoNode.collapsible = false;
+      repoNode.name = repo.first;
 
-        SerializedResponseNode isRunning;
-        isRunning.name = "running";
-        isRunning.value = repo.second->isRunning();
+      SerializedResponseNode queuesize;
+      queuesize.name = "size";
+      queuesize.value = repo.second->getRepoSize();
 
-        SerializedResponseNode isFull;
-        isFull.name = "full";
-        isFull.value = repo.second->isFull();
+      SerializedResponseNode isRunning;
+      isRunning.name = "running";
+      isRunning.value = repo.second->isRunning();
 
-        repoNode.children.push_back(queuesize);
-        repoNode.children.push_back(isRunning);
-        repoNode.children.push_back(isFull);
-        repositories.children.push_back(repoNode);
-      }
-      serialized.push_back(repositories);
+      SerializedResponseNode isFull;
+      isFull.name = "full";
+      isFull.value = repo.second->isFull();
+
+      repoNode.children.push_back(queuesize);
+      repoNode.children.push_back(isRunning);
+      repoNode.children.push_back(isFull);
+      repositories.children.push_back(repoNode);
     }
+    return repositories;
+  }
 
-    serialized.push_back(uptime);
+  SerializedResponseNode serializeUptime() {
+    SerializedResponseNode uptime;
 
+    uptime.name = "uptime";
     if (nullptr != monitor_) {
+      uptime.value = monitor_->getUptime();
+    } else {
+      uptime.value = "0";
+    }
+
+    return uptime;
+  }
+
+  SerializedResponseNode serializeComponents() {
+    SerializedResponseNode components_node(false);
+    components_node.name = "components";
+    if (monitor_ != nullptr) {
       auto components = monitor_->getAllComponents();
-      SerializedResponseNode componentsNode(false);
-      componentsNode.name = "components";
 
       for (auto component : components) {
```

Comment:
I know it's old code, but as we touch it - copy seems to be needless here.

Reply:
done [d2832cf](https://github.com/apache/nifi-minifi-cpp/pull/1032/commits/d2832cfcb3afa837e30e7bc95f496f0754d7603f)